CONTRIBUTING.md

```diff
@@ -0,0 +1,119 @@
+
+There are multiple ways to write tests: integration tests, documentation tests and unit tests.
+
+#### Integration tests
+
+[Integration tests][3] are contained in the `tests/` directory of the repository.
+
+The best strategy for writing a new integration test is to look at existing integration tests in the crate and follow the style.
+
+#### Documentation tests
+
+These are the [code examples][1] in the `doc/*.md` files and [README].
+
+Writing documentation tests is needed for better illustration of the added/altered capabilities for end-users of the crate.
+
+#### Unit tests
+
+[Unit tests][2] don't have much sense when it comes to macro testing. That's why they are rare beasts in the code of this repository. However, occasionally, they're very useful for testing some complicated properties locally (like correctness of syntax parsing, for example).
+
+
+### Review
+
+To get merged, any [PR] should be reviewed and approved by at least one of the active project maintainers, except the [PR] submitter, of course.
+
+Furthermore, __any `derive_more` community member is welcome to review any [PR].__
+
+All `derive_more` contributors who choose to review and provide feedback on [Pull Requests][PR] have a responsibility to both the project and the individual making the contribution. Reviews and feedback must be helpful, insightful, and geared towards improving the contribution as opposed to simply blocking it. If there are reasons why you feel the [PR] should not land, explain what those are. Do not expect to be able to block a [Pull Request][PR] from advancing simply because you say "No" without giving an explanation. Be open to having your mind changed. Be open to working with the contributor to make the [Pull Request][PR] better.
+
+When reviewing a [Pull Request][PR], the primary goals are for the codebase to improve and for the person submitting the request to succeed. Even if a [Pull Request][PR] doesn't land, the submitters should come away from the experience feeling like their effort was not wasted or unappreciated.
+
+#### Abandoned or stalled [Pull Requests][PR]
+
+If a [Pull Request][PR] appears to be abandoned or stalled, it's polite to first check with the contributor to see if they intend to continue the work before checking if they would mind if you took it over (especially if it just has nits left). When doing so, it's courteous to give the original contributor credit for the work they started (by preserving their name and email address with `Co-authored-by:` meta-data tag in the commit).
+
+
+### Merging
+
+All [PR]s are [squash]-merged to preserve the main history linear and meaningful.
+
+#### Commit message for a [squash] merge
+
+Commit message for a [squash] merge of the [PR] should mention its number and the number of the relevant [issue] (if it has the one).
```

Comment:
```suggestion
Commit message for a [squash] merge of the [PR] should mention its number and the number of the relevant [issue] (if it has the one). The commit body should contain the PR description. All of this is done automatically by Github.
```

Reply:
@JelteF I'm unsure of this suggestion. GitHub automatically not only adds the PR description, but also all the commit messages going to be squashed. Quite a lot of trash, as for me. I think it's better to write a laconic summary by hand.

Reply:
I set up the repo like this:
![image](https://user-images.githubusercontent.com/1162278/196945036-461e61ed-bd7c-4337-ab7d-be3acedf0e05.png)

Which is a new feature: https://github.blog/changelog/2022-08-23-new-options-for-controlling-the-default-commit-message-when-merging-a-pull-request/

Instead of combining all commit messages (indeed often resulting into trash), the default body is now the same as the PR description. The PR description is is usually a good description of what changes, and if it's not improvements to it can be suggested during review.